Environment Variable substitution not working

I am using following `docker-compose.yml` file
```yaml
$ cat docker-compose.yml 
version: "2"

services:
  mariadb:
    image: centos/mariadb
    ports:
      - "$DB_PORT"
    environment:
      MYSQL_ROOT_PASSWORD: $ROOT_PASS
      MYSQL_DATABASE: $DB_NAME
      MYSQL_PASSWORD: $DB_PASS
      MYSQL_USER: $DB_USER

  etherpad:
    image: centos/etherpad
    ports:
      - "80:9001"
    depends_on:
      - mariadb
    environment:
      DB_HOST: $DB_HOST
      DB_DBID: $DB_NAME
      DB_PASS: $DB_PASS
      DB_PORT: $DB_PORT
      DB_USER: $DB_USER
```

and following envs

```bash
$ cat envs 
DB_HOST=mariadb
ROOT_PASS=etherpad
DB_NAME=etherpad
DB_PASS=etherpad
DB_USER=etherpad
DB_PORT=3306
```

error
```
$ export $(cat envs)
$ ./kompose convert -o output
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x64e144]

goroutine 1 [running]:
panic(0x11ef080, 0xc820010140)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.Interpolate.func1(0xc8204acc88, 0x7, 0x0, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/interpolation.go:145 +0x84
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.parseVariable(0xc8204ac790, 0x8, 0x8, 0xc82015fbe8, 0x0, 0x0, 0x0, 0xc8204ba150)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/interpolation.go:36 +0x120
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.parseInterpolationExpression(0xc8204ac790, 0x8, 0x1, 0xc82015fbe8, 0x0, 0x0, 0xc82015f7a8, 0x4127d8)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/interpolation.go:74 +0x10f
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.parseLine(0xc8204ac790, 0x8, 0xc82015fbe8, 0x0, 0x0, 0x412672)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/interpolation.go:90 +0xbf
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.parseConfig(0xc8204ac718, 0x5, 0xc8204ac6f0, 0x7, 0xc82015f990, 0xc82015fbe8, 0x0, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/interpolation.go:110 +0x532
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.parseConfig(0xc8204ac718, 0x5, 0xc8204ac6f0, 0x7, 0xc82015fb68, 0xc82015fbe8, 0x0, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/interpolation.go:117 +0x1d3
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.Interpolate(0x0, 0x0, 0xc82015fd38, 0x0, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/interpolation.go:144 +0x1f3
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.MergeServicesV2(0xc8204638c0, 0x0, 0x0, 0x0, 0x0, 0xc820463880, 0x12, 0xc82049c000, 0x1d9, 0x3d9, ...)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/merge_v2.go:22 +0x12a
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config.Merge(0xc8204638c0, 0x0, 0x0, 0x0, 0x0, 0xc820463880, 0x12, 0xc82049c000, 0x1d9, 0x3d9, ...)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/config/merge.go:40 +0x24c
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/project.(*Project).load(0xc8204651e0, 0xc820463880, 0x12, 0xc82049c000, 0x1d9, 0x3d9, 0x0, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/project/project.go:178 +0xe5
github.com/skippbox/kompose/vendor/github.com/docker/libcompose/project.(*Project).Parse(0xc8204651e0, 0x0, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/docker/libcompose/project/project.go:86 +0x335
github.com/skippbox/kompose/cli/app.loadComposeFile(0xc820463880, 0x12, 0xc820347180, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/cli/app/app.go:733 +0x1b6
github.com/skippbox/kompose/cli/app.Convert(0xc820347180)
	/home/vagrant/work/src/github.com/skippbox/kompose/cli/app/app.go:1028 +0x4b9
github.com/skippbox/kompose/cli/command.ConvertCommand.func1(0xc820347180)
	/home/vagrant/work/src/github.com/skippbox/kompose/cli/command/command.go:30 +0x21
github.com/skippbox/kompose/vendor/github.com/urfave/cli.Command.Run(0x13eb378, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x159d920, 0x30, 0x0, ...)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/urfave/cli/command.go:179 +0x1399
github.com/skippbox/kompose/vendor/github.com/urfave/cli.(*App).Run(0xc82048ab00, 0xc82000a940, 0x4, 0x4, 0x0, 0x0)
	/home/vagrant/work/src/github.com/skippbox/kompose/vendor/github.com/urfave/cli/app.go:196 +0x137c
main.main()
	/home/vagrant/work/src/github.com/skippbox/kompose/cli/main/main.go:46 +0x5fe
```

version

```bash
$ ./kompose -v
kompose version 0.0.1-alpha (HEAD)
```

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -3,7 +3,7 @@
 	"GoVersion": "go1.6",
 	"GodepVersion": "v74",
 	"Packages": [
-		"./..."
+		"./cli/main/"
 	],
 	"Deps": [
 		{
@@ -44,77 +44,77 @@
 		},
 		{
 			"ImportPath": "github.com/docker/distribution",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/context",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/digest",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/manifest",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/manifest/schema1",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/manifest/schema2",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/reference",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/registry/api/errcode",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/registry/api/v2",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/registry/client",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/registry/client/auth",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/registry/client/transport",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/registry/storage/cache",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/registry/storage/cache/memory",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
 			"ImportPath": "github.com/docker/distribution/uuid",
-			"Comment": "v2.5.0-rc.1-22-g4e17ab5",
+			"Comment": "docs-v2.4.1-2016-06-28-6-g4e17ab5",
 			"Rev": "4e17ab5d319ac5b70b2769442947567a83386fbc"
 		},
 		{
@@ -434,73 +434,73 @@
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/config",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/docker",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/docker/builder",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/docker/client",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/docker/network",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/labels",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/logger",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/lookup",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/project",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/project/events",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/project/options",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/utils",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/version",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libcompose/yaml",
-			"Comment": "v0.2.0-186-ga12288b",
-			"Rev": "a12288bd636066e330b5d39197737cd04882d164"
+			"Comment": "v0.2.0-198-gced6fdd",
+			"Rev": "ced6fddc87aaf1bbf3c01df4530b6069508b8602"
 		},
 		{
 			"ImportPath": "github.com/docker/libtrust",

--- a/cli/app/app.go
+++ b/cli/app/app.go
@@ -20,6 +20,7 @@ import (
 	"fmt"
 	"math/rand"
 	"os"
+	"path/filepath"
 	"strconv"
 	"strings"
 
@@ -27,7 +28,9 @@ import (
 	"github.com/urfave/cli"
 
 	"github.com/docker/docker/api/client/bundlefile"
+	"github.com/docker/libcompose/config"
 	"github.com/docker/libcompose/docker"
+	"github.com/docker/libcompose/lookup"
 	"github.com/docker/libcompose/project"
 
 	"encoding/json"
@@ -741,6 +744,25 @@ func loadComposeFile(file string, c *cli.Context) KomposeObject {
 		file = "docker-compose.yml"
 	}
 	context.ComposeFiles = []string{file}
+
+	if context.ResourceLookup == nil {
+		context.ResourceLookup = &lookup.FileResourceLookup{}
+	}
+
+	if context.EnvironmentLookup == nil {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return KomposeObject{}
+		}
+		context.EnvironmentLookup = &lookup.ComposableEnvLookup{
+			Lookups: []config.EnvironmentLookup{
+				&lookup.EnvfileLookup{
+					Path: filepath.Join(cwd, ".env"),
+				},
+				&lookup.OsEnvLookup{},
+			},
+		}
+	}
 
 	// load compose file into composeObject
 	composeObject := project.NewProject(&context.Context, nil, nil)

